src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java

```diff
@@ -38,26 +41,40 @@ public class PageOCRPreparator {
-        log.debug("Class PageOCRPreparator started.");
-        String fileResult = "preparePage" + page + ".png";
-        String pagePDF = getPagePDF(document, page);
-        String rotatedPage = rotationImage.rotate(pagePDF);
-        Rectangle pageRectangle = getPageMainRectangle(document, page);
+    static final double HORIZONTAL_LINE_LENGTH = 700;
+    static final int HORIZONTAL_THRESHOLD = 5;
+    static final double ALL_LINES_THRESHOLD1 = 50;
+    static final double ALL_LINES_THRESHOLD2 = 200;
+    static final int ALL_LINES_APERTURESIZE = 3;
+    static final int ALL_LINES_RHO = 3;
+    static final int ALL_LINES_MAXLINEGAP = 5;
+    static final double VERTICAL_LINE_LENGTH = 100;
+
+    public String preparePage(PDDocument document, int pageNumber) throws IOException {
+        log.debug("Method pageOCRPreparator started.");
+        String fileResult = pageNumber + "_#6_prepared_page.png";
+        String pagePDF = getPagePDF(document, pageNumber);
+        String rotatedPage = rotationImage.rotateImage(pagePDF, pageNumber);
+        List<Rect> rects = extractListRect(rotatedPage);
+        Mat image = Imgcodecs.imread(rotatedPage);
+        final Mat result2 = cleanAllBesidesRects(image, rects, pageNumber);
+        Imgcodecs.imwrite(pageNumber + "_#3_cleanAllBesidesRects.png", result2);
+        java.util.List<Rectangle> pageRectanglesAllWords = getPageRectanglesAllWords(pageNumber + "_#3_cleanAllBesidesRects.png", pageNumber);
+        List<Rectangle> rectanglesWithSymbols = extractTextFromRectangle(pageNumber + "_#2_rotatedImage.png", pageRectanglesAllWords);
+        saveRectanglesOnImage(rectanglesWithSymbols, pageNumber + "_#2_rotatedImage.png", pageNumber);
+        Rectangle mainPageRectangle = findMainPageRectangle(rectanglesWithSymbols);
+        List<double[]> verticalLinesWithOpenCV = findVerticalLinesWithOpenCV(pageNumber + "_#2_rotatedImage.png");
+
+        double offset = getOffset(mainPageRectangle, verticalLinesWithOpenCV);
+        mainPageRectangle = offset > 0 ? createMainRect(mainPageRectangle, offset) : mainPageRectangle;
+
         Mat tableMat = Imgcodecs.imread(rotatedPage);
-        Mat matPage = cleanTableBorders(tableMat, pageRectangle);
-        Imgcodecs.imwrite(page + ".png", matPage);
-        java.util.List<Rectangle> pageRectanglesAllWords = getPageRectanglesAllWords(page + ".png");
-        List<Rectangle> rectanglesWithSymbols = extractTextFromRectangle(page + ".png", pageRectanglesAllWords);
-        saveRectanglesOnImage(rectanglesWithSymbols, page + ".png");
-        final Rectangle mainPageRectangle = findMainPageRectangle(rectanglesWithSymbols);
-        final Mat result = cleanTableBorders(matPage, mainPageRectangle);
+        final Mat result = cleanTableBorders(tableMat, mainPageRectangle, pageNumber);
         Imgcodecs.imwrite(fileResult, result);
-        log.debug("Class PageOCRPreparator finished.");
+        log.debug("Method PageOCRPreparator finished.");
```

Comment:
Consider using INFO level to track level for such cases. And add the name of the document and a page

Reply:
Done.